### PR TITLE
[core] Sort the expired partitions and add max_expires param

### PR DESCRIPTION
**Pull request by askwang**
### Purpose
- The result of expire_partitions is out of order, sort it.
- Add max_expires param to limit the expired partitions number at one time.

before:
image

after:
image

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java [hunk 1]
```diff
@@ -125,14 +133,16 @@ List<Map<String, String>> expire(LocalDateTime now, long commitIdentifier) {
...
+        List<List<String>> partValues = new ArrayList<>();
         for (PartitionEntry partition : strategy.selectExpiredPartitions(scan, expireDateTime)) {
             Object[] array = strategy.convertPartition(partition.partition());
-            Map<String, String> partString = strategy.toPartitionString(array);
-            expired.add(partString);
-            LOG.info("Expire Partition: {}", partString);
+            partValues.add(strategy.toPartitionValue(array));
         }
-        if (!expired.isEmpty()) {
+
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java:141]: Use follow code could init partValues with capacity.

List<PartitionEntry> partitionEntries = strategy.selectExpiredPartitions(scan, expireDateTime);
        List<List<String>> partValues = new ArrayList<>(partitionEntries.size());
        for (PartitionEntry partition : partitionEntries) {
            Object[] array = strategy.convertPartition(partition.partition());
            partValues.add(strategy.toPartitionValue(array));
        }

File: paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java [hunk 2]
```diff
@@ -153,4 +163,14 @@ private void deleteMetastorePartitions(List<Map<String, String>> partitions) {
                     });
         }
     }
+
+    private List<Map<String, String>> convertToPartitionString(List<List<String>> partValues) {
+        return partValues.stream()
+                .map(values -> String.join(",", values))
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java:169]: private final String xxx = ",";

File: paimon-core/src/main/java/org/apache/paimon/partition/PartitionExpireStrategy.java [hunk 3]
```diff
@@ -49,6 +50,14 @@ public Map<String, String> toPartitionString(Object[] array) {
         return map;
     }
 
+    public List<String> toPartitionValue(Object[] array) {
+        List<String> list = new ArrayList<>();
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/partition/PartitionExpireStrategy.java:54]: Change "List<String> list = new ArrayList<>();" to "List<String> list = new ArrayList<>(partitionKeys.size());".

File: paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java [hunk 4]
```diff
@@ -125,14 +133,16 @@ List<Map<String, String>> expire(LocalDateTime now, long commitIdentifier) {
...
-            expired.add(partString);
-            LOG.info("Expire Partition: {}", partString);
+            partValues.add(strategy.toPartitionValue(array));
         }
-        if (!expired.isEmpty()) {
+
+        List<Map<String, String>> expired = new ArrayList<>();
+        if (!partValues.isEmpty()) {
+            expired = convertToPartitionString(partValues);
+            LOG.info("Expire Partition: {}", expired);
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java:145]: Change "Expire Partition" to "Expire Partitions".

File: paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java [hunk 5]
```diff
@@ -125,14 +133,16 @@ List<Map<String, String>> expire(LocalDateTime now, long commitIdentifier) {
 
     private List<Map<String, String>> doExpire(
             LocalDateTime expireDateTime, long commitIdentifier) {
-        List<Map<String, String>> expired = new ArrayList<>();
+        List<List<String>> partValues = new ArrayList<>();
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java:136]: Change "partValues" to "expiredPartValues"?

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**